docs/configuration-files.rst

```diff
@@ -0,0 +1,191 @@
+    min_partition_sizes =
+        /      250 MiB
+        /usr   250 MiB
+
+    # Required minimal sizes of partitions.
+    # Specify a mount point and a size on each line.
+    req_partition_sizes =
+
+
+The supported sections and options are documented in the default configuration file.
+
+Default configuration file
+--------------------------
+
+The default configuration file provides a full default configuration of the installer.
+It defines and documents all supported sections and options. The file is located at
+``/etc/anaconda/anaconda.conf``:
+
+.. include:: ../data/anaconda.conf
+    :code: ini
+
+Product configuration files
+---------------------------
+
+The product configuration files allow to override some of the configuration options for
+specific products. The files are located at ``/etc/anaconda/product.d/``.
+
+.. note::
+
+    Anaconda previously used so called install classes for the product-specific configuration.
+    Install classes were completely removed and replaced by the product configuration files.
+
+Product identification
+^^^^^^^^^^^^^^^^^^^^^^
+
+The product is defined by a product name and optionally a variant name. For example, Fedora
+Server has a product name ``Fedora`` and a variant name ``Server``. See the terminology of
+`productmd <https://productmd.readthedocs.io/en/latest/terminology.html>`_.
+
+The product can be specified by the boot options ``inst.product`` and ``inst.variant``.
+Otherwise, it will be automatically loaded from the ``.buildstamp`` file on the installation
+media. This file is generated by ``lorax`` using the ``--product`` and ``--variant`` options.
+
+Based on the provided product and variant names, the installer will look up the right
+configuration file in the ``/etc/anaconda/product.d/`` directory.
+
+File structure
+^^^^^^^^^^^^^^
+
+Product configuration files have one or two extra sections that identify the product. The
+``[Product]`` section specifies the product and variant names of a product. The ``[Base Product]``
+section specifies the product and variant names of a base product if any. For example, Fedora is
+a base product of Fedora Server.
+
+.. note::
+
+    We are not going to support wildcards in product names. This used to be supported in install
+    classes and it caused a lot of problems. The product name can match unrelated configurations
+    and you cannot be sure which configuration is going to be used at the end.
+
+We support a simple inheritance of product configurations. The installer loads configuration files
+of the base products before it loads the configuration file of the specified product. For example,
+it will first load the configuration for Fedora and then the configuration for Fedora Server.
+
+.. note::
+
+    We are not going to support multiple inheritance. Are you kidding me? You can easily compare
+    two configuration files and verify the parts they are supposed to share. We do that in our
+    unit tests.
```

Comment:
I like it and I understand why you wrote it but I don't think it should be in our documentation.

Reply:
Fixed.